### PR TITLE
Allow args to be passed to label templates and begin label library

### PR DESCRIPTION
**Pull request by kgiszewski**
1. Allow args to be passed via the label templates.
2. Adds built-in label templates
3. Adds auto-detection of some property editors
4. Adds label and cache services

Usage:
`{{MyNamespace.MyFunction(text, {contentLength: 50})}}`

Automatic usage for:
- Umbraco.TinyMCEv3
- Imulus.UrlPicker
- MNTP
- MediaPicker

screen shot 2015-05-27 at 2 00 01 pm
screen shot 2015-05-27 at 2 01 40 pm

### CODE REVIEW

File: app/controllers/controller.js [hunk 1]
```diff
@@ -33,34 +33,50 @@
         if (template.length < 1)
             return fieldsetConfig.label;
 
-        var rgx = /{{(.*?)}}*/g;
+        var rgx = /{{([^)].*)}}/g;
```
Comment (kgiszewski) [inline app/controllers/controller.js]: @kjac Any chance you can test these changes so they don't break your previous PR for allowing multiple functions?

**Comment by kjac** [comment]
@kgiszewski I've tested a build of the latest label-library branch, and the alternative labels feature introduced in #247 still works - both with simple properties (e.g. ```{{title|subTitle}}```) and with custom functions (e.g. ```{{myFunction(title)|myFunction(subTitle)```).

This PR looks nice, good job :+1: 

A quick suggestion: Add dots (...) if the RTE value is cut by ```{{ArchetypeLabels.Rte(text)}}```?

**Comment by kgiszewski** [comment]
@kjac Awesome thank you! :+1:

**Comment by Nicholas-Westby** [comment]
@kjac Regarding your "add dots" suggestion, the ellipsis will be handled by this pull request that was recently merged (it's a CSS ellipsis, so it will apply the CSS based on how wide the screen is): https://github.com/imulus/Archetype/pull/272
ellipsis

Of course, it would still be valuable to add an ellipsis if the contentLength parameter is used to enforce a maximum length.

**Pull request merged**